imap_processing/ena_maps/ena_maps.py

```diff
@@ -252,7 +252,7 @@ def __init__(
         """Abstract method to initialize the pointing set object."""
         self.spice_reference_frame = spice_reference_frame
 
-        if isinstance(dataset, (str, Path)):
+        if isinstance(dataset, str | Path):
```

Comment:
Can you ignore rule [UP038](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/#non-pep604-isinstance-up038)
It is deprecated to keep the standard checking style in place. That will save a bit of this churn.